VmAgent.Core.UnitTests/SessionHostContainerConfigurationTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Azure.Gaming.VmAgent.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using VmAgentCoreInterface = Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+using Docker.DotNet;
+using Microsoft.Azure.Gaming.AgentInterfaces;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using System.Collections.Generic;
+using System.IO;
+using System;
+using Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+
+namespace VmAgent.Core.UnitTests
+{
+    [TestClass]
+    public class SessionHostContainerConfigurationUnitTests
+    {
+        // Directory Variables
+        private string _root = "C:\\";
+        private string _VmDirectoryRoot = "root";
+        private string _VmDirectoryContainerRoot = "/data";
+    
+        // Environment Variables
+        private const string TitleIdEnvVariable = "PF_TITLE_ID";
+        private const string BuildIdEnvVariable = "PF_BUILD_ID";
+        private const string RegionEnvVariable = "PF_REGION";
+        private const string LogsDirectoryEnvVariable = "PF_SERVER_LOG_DIRECTORY";
+        private const string ServerInstanceNumberEnvVariable = "PF_SERVER_INSTANCE_NUMBER";
+
+        private static Guid TestTitleIdUlong = Guid.NewGuid();
+        private static Guid TestBuildId = Guid.NewGuid();
+        private static Guid TestRegion = Guid.NewGuid();
+        private static string TestLogFolderId = "TestLogFolderId";
+        private static string TestPublicIpV4Address = "127.0.0.1";
+
+        private VmConfiguration _vmConfiguration;
+        private MultiLogger _logger;
+        private Mock<VmAgentCoreInterface.ISystemOperations> _systemOperations;
+        private Mock<ISessionHostManager> _sessionHostManager;
+        private Mock<IDockerClient> _dockerClient;
+        private SessionHostsStartInfo _sessionHostsStartInfo;
+
+        [TestInitialize]
+        public void BeforeEachTest()
+        {
+            string AssignmentId = $"{TestTitleIdUlong}:{TestBuildId}:{TestRegion}";
+
+            _logger = new MultiLogger(NullLogger.Instance, new TelemetryClient(TelemetryConfiguration.CreateDefault()));
+            _systemOperations = new Mock<Microsoft.Azure.Gaming.VmAgent.Core.Interfaces.ISystemOperations>();
+            _dockerClient = new Mock<IDockerClient>();
+            _sessionHostManager = new Mock<ISessionHostManager>();
+            _vmConfiguration = new VmConfiguration(56001, "vmid", new VmDirectories(_VmDirectoryRoot), true);
+            //string rootOutputFolder = Path.Combine(settings.OutputFolder, "PlayFabVmAgentOutput", DateTime.Now.ToString("s").Replace(':', '-'));
```

Comment:
do we need this?

Reply:
Does it mean we don't need to create mock objects and just need to create it when we pass it as an argument?

Reply:
no, I'm sorry, my question was if we need the commented out code or we should remove it. Thanks!